src/neon.cc

```diff
@@ -0,0 +1,80 @@
+/*  ====================================================================================================================
+    Copyright (C) 2020 Eaton
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+    ====================================================================================================================
+*/
+
+#include "neon.h"
+#include <fty/string-utils.h>
+#include <neon/ne_request.h>
+#include <neon/ne_session.h>
+#include <neon/ne_xml.h>
+#include <pack/visitor.h>
+#include <iostream>
+
+namespace neon {
+
+Neon::Neon(const std::string& address, uint16_t port, uint16_t timeout)
+    : m_session(ne_session_create("http", address.c_str(), port), &closeSession)
```

Comment:
would be good to make scheme (http) also configurable, to be future proof. Though can be addressed later...